pkg/asset/installconfig/aws/permissions.go

```diff
@@ -493,3 +504,35 @@ func includesExistingInstanceProfile(installConfig *types.InstallConfig) bool {
 
 	return len(mpool.IAMProfile) > 0
 }
+
+// includesCreateInstanceProfile checks if at least one instance profile will be created by the Installer.
+func includesCreateInstanceProfile(installConfig *types.InstallConfig) bool {
```

Comment:
Is this the opposite of includesExistingInstanceProfile() ?

Reply:
It's not and that's why I have the two. Imagine a configuration where master has a BYO profile but worker does not. In that case, both `includesCreateInstanceProfile` and `includesExistingInstanceProfile` should return `true` as we need all permissions to create and use existing profiles.